File: server/router/task/daily_task.go
package task

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type DailyTaskRouter struct{}

func (s *DailyTaskRouter) InitDailyTaskRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	dailyTaskRouter := Router.Group("dailyTask").Use(middleware.OperationRecord())
	{
		dailyTaskRouter.POST("createDailyTask", dailyTaskApi.CreateDailyTask)
		dailyTaskRouter.DELETE("deleteDailyTask", dailyTaskApi.DeleteDailyTask)
		dailyTaskRouter.DELETE("deleteDailyTaskByIds", dailyTaskApi.DeleteDailyTaskByIds)
		dailyTaskRouter.PUT("updateDailyTask", dailyTaskApi.UpdateDailyTask)
	}

	dailyTaskRouterWithoutRecord := Router.Group("dailyTask")
	{
		dailyTaskRouterWithoutRecord.GET("findDailyTask", dailyTaskApi.FindDailyTask)
		dailyTaskRouterWithoutRecord.GET("getDailyTaskList", dailyTaskApi.GetDailyTaskList)
		dailyTaskRouterWithoutRecord.GET("listActiveTaskList", dailyTaskApi.ListActiveTaskList)
	}

	dailyTaskRouterWithoutAuth := PublicRouter.Group("dailyTask")
	{
		_ = dailyTaskRouterWithoutAuth
	}
}

File: web/src/api/task/daily_task.js
import service from '@/utils/request'

export const createDailyTask = (data) => {
  return service({
    url: '/dailyTask/createDailyTask',
    method: 'post',
    data
  })
}

export const deleteDailyTask = (params) => {
  return service({
    url: '/dailyTask/deleteDailyTask',
    method: 'delete',
    params
  })
}

export const deleteDailyTaskByIds = (params) => {
  return service({
    url: '/dailyTask/deleteDailyTaskByIds',
    method: 'delete',
    params
  })
}

export const updateDailyTask = (data) => {
  return service({
    url: '/dailyTask/updateDailyTask',
    method: 'put',
    data
  })
}

export const findDailyTask = (params) => {
  return service({
    url: '/dailyTask/findDailyTask',
    method: 'get',
    params
  })
}

export const getDailyTaskList = (params) => {
  return service({
    url: '/dailyTask/getDailyTaskList',
    method: 'get',
    params
  })
}

export const listActiveTaskList = () => {
  return service({
    url: '/dailyTask/listActiveTaskList',
    method: 'get'
  })
}